### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by tok-kkk**
- Split the big store.go file into specific files.
- Migrate the IterableStore from `lightnode` to this package.
- Add more detailed comments with the interface 
- Add tests. (test coverage above 95%)

### CODE REVIEW

File: badgerDB.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
+package store
```
Comment (loongy) [inline badgerDB.go]: This file should be named `bdb.go` or `badgerdb.go` to stay consistent with standard capitalisation.

File: levelDB.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package store
```
Comment (loongy) [inline levelDB.go]: This file should be named `ldb.go` or `leveldb.go` to stay consistent with standard capitalisation.

File: cache.go [hunk 3]
```diff
@@ -0,0 +1,233 @@
+package store
```
Comment (loongy) [inline cache.go]: This file should be named `mdb.go` or `memdb.go` to provide better semantic meaning for this implementation that is consistent with the other implementations.

**Review by loongy — CHANGES_REQUESTED** [review]
Overall, this is a good first implementation of the KV storage library. However, I have pushed some changes that implement a more flexible architecture. The major changes:

- Databases are separated from stores. Databases only understand the storage of bytes, whereas stores also understand the encoding. For example: badgerdb, leveldb, and memdb are all databases, whereas json, and gob are both considered stores.
- Caching has been moved into its own package and is, itself, an implementation of a store. It wraps other stores and deletes key-value tuples that have not been accessed for at least the TTL duration.
- Support for gob encoding/decoding has been added.

Some things that now need doing:
- [ ] Support for LevelDB iterators.
- [ ] Fix test suites.

**Comment by tok-kkk** [comment]
Thanks. I'll go over the changes and finish things left. I'll let you know when it's ready for another review.

**Review by loongy — APPROVED** [review]

**Pull request merged**